src/mapping/grouping/pairwise_comparison.jl

```diff
@@ -0,0 +1,108 @@
+
+
+"""
+    GroupViaPairwiseComparison(; optimal_radius_method::Real, kwargs...)
+
+Initialize a struct that contains instructions on how to group features in
+[`AttractorsViaFeaturizing`](@ref). `GroupViaPairwiseComparison` groups features and
+identifies clusters by considering the pairwise distance (using `distance_metric`) between
+features. 
+
+## Keyword arguments
+    
+* `distance_threshold`: a real number defining the maximum distance two features can be to
+  be considered in the same cluster - above the threshold, features are different. This
+  value simply needs to be large enough to differentiate clusters.
+* `distance_metric = Euclidean()`: A metric to be used in the clustering. It can be any
+  function `f(a, b)` that returns the distance between any type of data structure (usually
+  vectors or matrices of reals). Needs to be consistent with the `featurizer` function. All
+  metrics from Distances.jl can be used here.
+* `rescale_features = true`: if true, rescale each dimension of the extracted features
+  separately into the range `[0,1]`. This typically leads to more accurate clustering.
+
+## Description
+This algorithm assumes that the the features well-separated into distinct clouds, with the
+maximum radius of the cloud controlled by `distance_threshold`. Since the systems are
+deterministic, this is achievable with a good-enough `featurizer` function, by removing
+transients, and running the trajectories for sufficiently long. It then considers that
+features belong to the same attractor when their pairwise distance, computed using
+`distance_metric`, is smaller or equal than `distance_threshold`, and belong to different
+attractors when the distance is bigger than the threshold. Attractors correspond to each
+grouping of similar features. In this way, the key parameter `distance_threshold` is
+simply the amount of variations permissible in the features. If they are well-chosen, the
+value can be relatively small and does not need to be fine tuned.
+"""
+struct GroupViaPairwiseComparison{R<:Real, M} <: GroupingConfig
+    optimal_radius_method::R
+    distance_metric::M
+    rescale_features::Bool 
+end
+
+function GroupViaPairwiseComparison(;
+        optimal_radius_method, #impossible to set a good default value, depends on the features
+        distance_metric=Euclidean(), rescale_features=false, 
+    )
+    return GroupViaPairwiseComparison(
+        optimal_radius_method,
+        distance_metric, rescale_features,
+    )
+end
+
+function group_features(
+    features, config::GroupViaPairwiseComparison;
+    kwargs...
+)
+    if config.rescale_features
+        features = _rescale_to_01(features)
+    end
+    ϵ_optimal = _extract_ϵ_optimal(features, config)
+    labels = _cluster_features_into_labels(features, config, ϵ_optimal; kwargs...)
+    return labels
+end
+
+function _extract_ϵ_optimal(features, config::GroupViaPairwiseComparison)
+    return config.optimal_radius_method
+end
```

Comment:
I am confused about the source code here... Why do you need this function? And why is something called `ε_optimal` at some places and `distance_threshold` at others? Are we doing any clustering here after all? No right? There isn't any `ε_optimal`? Or is there. :S

Reply:
That was stupid of me, I'll change everything to `distance_threshold` as it should be.